server/src/test/java/org/opensearch/gateway/ShardsBatchGatewayAllocatorTests.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.gateway;
+
+import org.opensearch.action.admin.cluster.settings.ClusterUpdateSettingsResponse;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.test.OpenSearchSingleNodeTestCase;
+
+import static org.opensearch.gateway.ShardsBatchGatewayAllocator.GATEWAY_ALLOCATOR_BATCH_SIZE;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.equalTo;
+
+public class ShardsBatchGatewayAllocatorTests extends OpenSearchSingleNodeTestCase {
```

Comment:
test for batchAllocator are in GatewayAllocator itself.

Reply:
wanted to introduce a new class which extends `OpenSearchSingleNodeTestCase` as this class exposes a `client()` function so it is easy to call admin APIs and is easier to write the tests

`GatewayAllocatorTests` extends `OpenSearchAllocationTestCase ` which does not expose such a method to call admin APIs

In any case, thought that we should have a class `ShardsBatchGatewayAllocatorTests` for `ShardsBatchGatewayAllocator` class since it doesn't exist already